docs/spec_draft.md

```diff
@@ -343,6 +344,67 @@ logical operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.batch_norm_training
+
+### Semantics
+
+Computes mean and variance across batch and spatial dimensions and normalizes
+the `operand` tensor, for each feature in the `feature_index` dimension and
+produces `output`, `batch_mean` and `batch_var` tensors. Refer the
+`Batch Normalizing Transform` algorithm from the
+[Batch Normalization](https://arxiv.org/abs/1502.03167) paper for the detailed
+algorithm.
+
+### Inputs
+
+| Name            | Type                                        |
+|-----------------|---------------------------------------------|
+| `operand`       | tensor of floating-point type               |
+| `scale`         | 1-dimensional tensor of floating-point type |
+| `offset`        | 1-dimensional tensor of floating-point type |
+| `epsilon`       | constant of type `f32`                      |
+| `feature_index` | constant of type `si64`                     |
+
+### Outputs
+
+| Name         | Type                                        |
+|--------------|---------------------------------------------|
+| `output`     | tensor of floating-point type               |
+| `batch_mean` | 1-dimensional tensor of floating-point type |
+| `batch_var`  | 1-dimensional tensor of floating-point type |
+
+### Constraints
+
+  * (C1) 0 $\le$ `feature_index` $\lt$ rank(`operand`).
+  * (C2) size(`scale`) $=$ `dim(operand, feature_index)`.
+  * (C3) size(`offset`) $=$ `dim(operand, feature_index)`.
+  * (C4) size(`batch_mean`) $=$ `dim(operand, feature_index)`.
+  * (C5) size(`batch_var`) $=$ `dim(operand, feature_index)`.
+  * (C6) `operand` and `output` have the same type.
+
+### Examples
+
+```mlir
+// %operand: [
+//            [[1.0, 2.0], [3.0, 4.0]],
+//            [[3.0, 4.0], [1.0, 2.0]]
+//           ]
+// %scale: [1.0, 1.0]
+// %offset: [1.0, 1.0]
+%results:3 = "stablehlo.batch_norm_training"(%operand, %scale, %offset) {
+  epsilon = 0.0 : f32,
+  feature_index = 2 : i64
+} : (tensor<2x2x2xf32>, tensor<2xf32>, tensor<2xf32>) -> (tensor<2x2x2xf32>, tensor<2xf32>, tensor<2xf32>)
+// %results#0: [
+//              [[0, 0], [2, 2]],
+//              [[2, 2], [0, 0]]
+//             ]
+// %results#1: [2, 3]
+// %results#2: [1, 1]
```

Comment:
Similarly to https://github.com/openxla/stablehlo/pull/377#discussion_r1013464149, since we're using floating-point notation above, let's use it for the example output as well.

Reply:
The outputs of the interpreter were without decimal point so I removed to be precise, will add it again.